exaudfclient/docs/script_options_requirments.md

```diff
@@ -0,0 +1,372 @@
+Database Administrators manage the database environment and ensure the efficient execution of UDFs within Exasol, configuring and overseeing script execution.
+
+### Data Scientist
+Data Scientists develop and deploy UDFs in languages such as Java, Python, or R to process and analyze data within Exasol.
+
+### Developer
+Developers who integrates the Script Options parser library into other software.
+
+## Features
+
+This section lists the key features of the new UDF Client Script Options parser which you would highlight in a product leaflet.
+
+### General Script Options Parsing
+`feat~general-script-options-parsing~1`
+
+Script Options must be parsed according to a given syntax definition.
+Developers can add additional Options in an easy and consistent way.
+
+Needs: req
+
+### Java-specific Script Options
+`feat~java-specific-script-options~1`
+
+The parser must process all Java-specific options correctly.
+
+Needs: req
+
+## High-level Requirements
+
+This section details the high-level requirements for the new parser system, linked to the features listed above.
+
+### General Script Options Parsing
+`req~general-script-options-parsing~1`
+
+The parser must correctly identify and handle Script Options with the syntax `%<optionKey><white spaces><optionValue>;`.
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+### White Spaces
+`req~white-spaces~1`
+
+The parser must treat the following list of white spaces as token separator:
+|======================================================= 
+| Name         | C syntax      | ASCII Dec | ASCII Hex | 
+| tabulator    | '\t'          | 9         | 0x09      |
+| vertical tab | '\v'          | 11         | 0x0b     |
+| form feed    | '\f'          | 12         | 0x0c     |
+| space        | ' '           | 30        | 0x20      |     
+|======================================================= 
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+
+### Leading White Spaces Options Parsing
+`req~leading-white-spaces-script-options-parsing~1`
+
+The parser must accept Script Options for lines starting with any number of white space characters before the Script Options.
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+Depends:
+ - `req~white-spaces~1`
+
+### Ignore anything which is not a Script Option
+`req~ignore-none-script-options~1`
+
+If there is any character in front of a Script Option which is not a white space, the parser must ignore the option(s). 
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+Depends:
+ - `req~white-spaces~1`
+
+### Multiple Line Script Options Parsing
+`req~multiple-lines-script-options-parsing~1`
+
+The parser must recognize Script Options at any line in the given script code. This is especially important because the `%import` option (see requirement `req~java-import-option-replace-referenced-scripts~1`) might replace options with Java code in the final script code.
```

Comment:
```suggestion
The parser must recognize Script Options at any line in the given script code. 

Rationale: This is especially important because the `%import` option (see requirement `req~java-import-option-replace-referenced-scripts~1`) might replace options with Java code in the final script code.
```